DOCS-7349 external lib (#187)
mulesoft/docs-mule-sdk: modules/ROOT/pages/external-libs.adoc

* `NATIVE`: Native library dependency
* `JAR`: JAR file dependency
* `DEPENDENCY`: External dependency (for example, an obtained dependency that is specified in the `pom.xml` if you are using Maven)

[NOTE]
=======

> For custom connectors, all Maven dependencies must be accessible from Anypoint Exchange. Maven dependencies published to private or public repositories are unaccessible from Anypoint Exchange.

Reviewer: There is a bit of debate over "unaccessible" and "inaccessible" (with the latter being preferred). I have this solution:
```suggestion
For custom connectors, all Maven dependencies must be accessible from Anypoint Exchange. Maven dependencies published to private or public repositories are not accessible from Anypoint Exchange.
```
lol: when in doubt, reword for simplicity!

> You can publish Maven dependencies before publishing your custom connector through one of the following options:

Reviewer: ```suggestion
You can publish Maven dependencies before publishing your custom connector by using one of the following options:
```

> * Publish the Maven dependencies on Anypoint Exchange to make them available for your Anypoint Platform organization.

Reviewer: ```suggestion
* Publish the Maven dependencies on Anypoint Exchange to make them available to your Anypoint Platform organization.
```

* Publish the Maven dependencies on the https://maven.apache.org/repository/guide-central-repository-upload.html[Maven Central repository] to make them publicly available.

> MuleSoft repositories are not open for publishing.

Reviewer: "are not open for" = cannot be?
